internal/k8s/status.go

```diff
@@ -177,23 +180,27 @@ func (irs *StatusWriter) setIngressRouteStatus(existing, updated *ingressroutev1
 }
 
 func (irs *StatusWriter) setHTTPProxyStatus(existing, updated *projcontour.HTTPProxy) error {
-	existingBytes, err := json.Marshal(existing)
-	if err != nil {
-		return err
-	}
-	// Need to set the resource version of the updated endpoints to the resource
-	// version of the current service. Otherwise, the resulting patch does not
-	// have a resource version, and the server complains.
-	updated.ResourceVersion = existing.ResourceVersion
-	updatedBytes, err := json.Marshal(updated)
+
+	usUpdated, err := toUnstructured(updated, projcontour.SchemeGroupVersion.WithKind("HTTPProxy"))
 	if err != nil {
 		return err
 	}
-	patchBytes, err := jsonpatch.CreateMergePatch(existingBytes, updatedBytes)
+
+	_, err = irs.Client.Resource(projcontour.HTTPProxyGVR).Namespace(existing.GetNamespace()).UpdateStatus(context.TODO(), usUpdated, metav1.UpdateOptions{})
+	return err
+}
+
+func toUnstructured(obj runtime.Object, gvk schema.GroupVersionKind) (*unstructured.Unstructured, error) {
+
+	obj.GetObjectKind().SetGroupVersionKind(gvk)
+	objMsi, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	_, err = irs.Client.Resource(projcontour.HTTPProxyGVR).Namespace(existing.GetNamespace()).Patch(context.TODO(), existing.GetName(), types.MergePatchType, patchBytes, metav1.PatchOptions{})
-	return err
+	us := &unstructured.Unstructured{}
+	us.SetUnstructuredContent(objMsi)
```

Comment:
I think that the default scheme can do this conversion directly without needing to pass in the GVK, because we register all our types with it:
```
        u := &unstructured.Unstructured{}

        if err := scheme.Scheme.Convert(obj, u, nil); err != nil {
                log.Fatalf("conversion to unstructured failed: %s", err)
        }

        return u
```

Might need to check into the overlap with `k8s.UnstructuredConverter`.

Reply:
The UnstructuredConverter currently only converts *from* unstructured. Makes sense for it to do both ways though. I'll refactor accordingly, that way I'll move the logic where it should be and make it more available.